### ISSUE TITLE
M1 Mac Monterey 12.3 build failed with error "cannot cleanup"

### ISSUE THREAD

**Issue by jiadesen** [electron-userland/electron-builder#6723]
<!--
issue_labeler_regex_version=0
--!>

<!-- Which version of electron-builder are you using? -->
<!-- Please always try to use latest version before reporting any issue. -->
* **Electron-Builder Version**: ^22.14.13

<!-- Which version of node are you using? -->
* **Node Version**: v16.14.0

<!-- Which version of electron are you using? -->
<!-- Please verify if the error is electron related before reporting any issues. -->
* **Electron Version**: ^17.1.2
* **Electron Type (current, beta, nightly)**: current
<!-- Which version of electron-updater are you using (if applicable)? -->

<!-- For which target are you building for? -->
* **Target**: macos

<!-- Enter your issue details below this comment. -->
<!-- If you want, you can donate to increase issue priority (https://www.electron.build/donate) -->

Build log:

```
> electron-builder -m --arm64 --x64

  • electron-builder  version=22.14.13 os=21.4.0
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=output/app/builder-effective-config.yaml
  • packaging       platform=darwin arch=x64 electron=17.1.2 appOutDir=output/app/mac
  • skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, it could cause some undefined behaviour, e.g. macOS localized description not visible, see https://electron.build/code-signing allIdentities=     0 identities found
                                                Valid identities only
     0 valid identities found
  • building        target=DMG arch=x64 file=output/app/filename-x64.dmg
  • Detected arm64 process, HFS+ is unavailable. Creating dmg with APFS - supports Mac OSX 10.12+
  • packaging       platform=darwin arch=arm64 electron=17.1.2 appOutDir=output/app/mac-arm64
  • skipped macOS application code signing  reason=cannot find valid "Developer ID Application" identity or custom non-Apple code signing certificate, it could cause some undefined behaviour, e.g. macOS localized description not visible, see https://electron.build/code-signing allIdentities=     0 identities found
                                                Valid identities only
     0 valid identities found
  • building        target=DMG arch=arm64 file=output/app/filename-arm64.dmg
  • Detected arm64 process, HFS+ is unavailable. Creating dmg with APFS - supports Mac OSX 10.12+
  ⨯ Cannot cleanup: 

Error #1 --------------------------------------------------------------------------------
Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:406:5)
    at ChildProcess.errorhandler (node:child_process:418:5)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

Error #2 --------------------------------------------------------------------------------
Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:406:5)
    at ChildProcess.errorhandler (node:child_process:418:5)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)  failedTask=build stackTrace=Error: Cannot cleanup: 
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Error #1 --------------------------------------------------------------------------------
Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:406:5)
    at ChildProcess.errorhandler (node:child_process:418:5)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Error #2 --------------------------------------------------------------------------------
Error: Exit code: ENOENT. spawn /usr/bin/python ENOENT
    at /Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/util.ts:133:18
    at exithandler (node:child_process:406:5)
    at ChildProcess.errorhandler (node:child_process:418:5)
    at ChildProcess.emit (node:events:520:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:478:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    at throwError (/Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/asyncTaskManager.ts:88:11)
    at checkErrors (/Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/asyncTaskManager.ts:53:9)
    at AsyncTaskManager.awaitTasks (/Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/asyncTaskManager.ts:67:7)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Packager.doBuild (/Users/xxx/electron-project/node_modules/.pnpm/app-builder-lib@22.14.13/node_modules/app-builder-lib/src/packager.ts:453:5)
    at Object.executeFinally (/Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/promise.ts:12:14)
    at Packager._build (/Users/xxx/electron-project/node_modules/.pnpm/app-builder-lib@22.14.13/node_modules/app-builder-lib/src/packager.ts:376:31)
    at Packager.build (/Users/xxx/electron-project/node_modules/.pnpm/app-builder-lib@22.14.13/node_modules/app-builder-lib/src/packager.ts:337:12)
    at Object.executeFinally (/Users/xxx/electron-project/node_modules/.pnpm/builder-util@22.14.13/node_modules/builder-util/src/promise.ts:12:14)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by dwrtz** [comment]
I upgraded to macos version 12.3 yesterday and now I'm seeing this issue

**Comment by SerhiiGadzhilov** [comment]
@jiadesen @dwrtz 
Looks like this is duplicate of https://github.com/electron-userland/electron-builder/issues/6606
It has already been fixed under https://github.com/electron-userland/electron-builder/pull/6617
It should work with version 23.0.2
As alternative solution you can install Python 2 manually and set environment variable PYTHON_PATH

**Comment by jiadesen** [comment]
> @jiadesen @dwrtz Looks like this is a duplicate of #6606 It has already been fixed under #6617 It should work with version 23.0.2 As alternative solution you can install Python 2 manually and set environment variable PYTHON_PATH

23.0.2 works very well, thanks.

**Comment by mmaietta** [comment]
Closing per https://github.com/electron-userland/electron-builder/issues/6723#issuecomment-1068424361
